src/components/questionnaire/CodeInput.tsx

```diff
@@ -1,16 +1,48 @@
+import { useNavigation } from '@react-navigation/native';
 import React, { useState } from 'react';
-import { StyleSheet, TextInput, TouchableOpacity, View } from 'react-native';
+import {
+  Alert,
+  Platform,
+  StyleSheet,
+  TextInput,
+  ToastAndroid,
+  TouchableOpacity,
+  View,
+} from 'react-native';
 import Icon from 'react-native-vector-icons/MaterialIcons';
 import ACCESSIBILITY_STRINGS from '../../assets/accessibilityStrings';
 import COLORS from '../../assets/colors';
 import FONTS from '../../assets/fonts';
+import { getAllQuestionnaireDataByCode } from '../../data/api/Questionnaire';
+import ParticipantCode from '../../data/localStorage/ParticipantCode';
+import { Questionnaire } from '../../models/Questionnaire';
 
 const CodeInput = () => {
-  const [code, setCode] = useState<String>('');
+  const [code, setCode] = useState<string>('');
+  const navigation = useNavigation();
 
-  const handleCodeEntered = () => {
-    //* TODO: make code entered function
-    return code;
+  const handleCodeEntered = async () => {
+    let questionnaireResponse = await getAllQuestionnaireDataByCode(code);
+    if (questionnaireResponse.status === 200) {
+      const questionnaire: Questionnaire = questionnaireResponse.data;
+      await ParticipantCode.saveCurrentParticipantCodeToLocalStorage(code);
+      await ParticipantCode.addQuestionnaireInLocalStorage({
+        code: code,
+        name: questionnaire.title,
+      });
+      // @ts-ignore next-line
+      navigation.navigate('Questionnaire', {
+        title: questionnaire.title,
+        questionnaire: questionnaire,
+      });
+    } else {
+      const msg = 'Vragenlijst niet gelukt op te halen';
+      if (Platform.OS === 'android') {
```

Comment:
Dit werkt niet, als de API request niet successvol is wordt er een error gegooid namelijk